contracts/ERC20MultiDelegate.sol

```diff
@@ -0,0 +1,251 @@
+     * @param delegatees List of delegatee addresses
+     */
+    function withdrawMulti(address[] calldata delegatees) external {
+        uint256 delegateesLength = delegatees.length;
+
+        require(
+            delegateesLength > 0,
+            "WithdrawMulti: You should pick at least one delegatee"
+        );
+
+        uint256[] memory delegates = new uint256[](delegateesLength);
+        uint256[] memory amounts = new uint256[](delegateesLength);
+
+        for (uint256 index = 0; index < delegateesLength; index++) {
+            address delegatee = delegatees[index];
+            uint256 amount = getBalanceForDelegatee(delegatee);
+            delegates[index] = uint256(uint160(delegatee));
+            amounts[index] = amount;
+
+            address proxyAddress = getProxyContractAddress(delegatee);
+            transferVotingPower(proxyAddress, msg.sender, amount);
+        }
+
+        burnBatch(msg.sender, delegates, amounts);
+    }
+
+    function setUri(string memory uri) external onlyOwner {
+        _setURI(uri);
+    }
+
+    function createProxyDelegatorAndTransfer(
+        address delegatee,
+        uint256 amount
+    ) internal {
+        address proxyAddress = deployProxyDelegatorIfNeeded(delegatee);
+        token.transferFrom(msg.sender, proxyAddress, amount);
+    }
+
+    function transferBetweenDelegators(
+        address from,
+        address to,
+        uint256 amount
+    ) internal {
+        address proxyAddressFrom = getProxyContractAddress(from);
+        address proxyAddressTo = getProxyContractAddress(to);
+        token.transferFrom(proxyAddressFrom, proxyAddressTo, amount);
+    }
+
+    function deployProxyDelegatorIfNeeded(
+        address delegatee
+    ) internal returns (address) {
+        (address proxyAddress, bytes32 salt) = retrieveProxyContractAddress(
+            token,
+            delegatee
+        );
+        new ERC20ProxyDelegator{salt: salt}(token, delegatee);
+        return proxyAddress;
+    }
+
+    function getBalanceForDelegatee(
+        address delegatee
+    ) internal view returns (uint256) {
+        return ERC1155(this).balanceOf(msg.sender, uint256(uint160(delegatee)));
+    }
+
+    function getProxyContractAddress(
+        address delegatee
+    ) internal view returns (address) {
+        (address proxyAddress, ) = retrieveProxyContractAddress(
+            token,
+            delegatee
+        );
+        return proxyAddress;
+    }
+
+    function transferVotingPower(
+        address from,
+        address to,
+        uint256 amount
+    ) internal {
+        require(
+            token.transferFrom(from, to, amount),
+            "Failed to transfer voting power"
+        );
+    }
+
+    function mintBatch(
+        address account,
+        uint256[] memory ids,
+        uint256[] memory amounts
+    ) internal {
+        _mintBatch(account, ids, amounts, "");
```

Comment:
Are these functions necessary?

Reply:
moved it into a separate method for modularity, in case we may need to perform additional checks or calls. can just call the private method instead if it seem unnecessary.

Reply:
I think unless it needs to be called from multiple places, you should skip having a separate internal method.